tests/functional/test_proxy.py

```diff
@@ -0,0 +1,35 @@
+import contextlib
+import subprocess
+import sys
+from typing import Iterator
+
+import pytest
+
+from tests.lib import PipTestEnvironment
+from tests.lib.path import Path
+
+
+@contextlib.contextmanager
+def run_proxy_server() -> Iterator[None]:
+    proc = subprocess.Popen([sys.executable, "-m", "proxy"])
+    yield
+    proc.kill()
+
+
+@pytest.mark.network
+def test_download_over_http_proxy(script: PipTestEnvironment) -> None:
+    """
+    It should download (in the scratch path) and not install if requested.
+    """
+    with run_proxy_server():
+        result = script.pip(
+            "download",
+            "--proxy",
+            "http://127.0.0.1:8899",
+            "--trusted-host",
+            "127.0.0.1",
```

Comment:
This test would pass even if these options were changed to be no-ops. Ideally, the test should fail if the proxy server isn't used.

Reply:
Yes I realized this also, I'm experimenting with disabling the network first, I tried a few different options including https://github.com/best-doctor/pytest_network  but I am struggling to get the test to fail. I think there are some nuances to pytest I do not understand correctly. I am going to close the PR for now until I understand the situation better, sorry for using up your time.

Reply:
Why not have the proxy write something to a file (essentially just "Hi, I was used") and then check the existence of the file? Or just set the proxy to write log output and check that, if that's easier?

Reply:
@pfmoore That makes a lot of sense. Unfortunately it seems that "proxy.py" has issues outputting access logs on Windows, I've raised an issue over there to see if there's an easy fix.

In the mean time I am checking if I could use it's plugin architecture to extract the details in another way or find another proxy tool that gives me the details more simply.